src/alu_instruction.rs

```diff
@@ -0,0 +1,43 @@
+pub(crate) enum ArmModeAluInstruction {
+    AND = 0x0,
+    EOR = 0x1,
+    SUB = 0x2,
+    RSB = 0x3,
+    ADD = 0x4,
+    ADC = 0x5,
+    SBC = 0x6,
+    RSC = 0x7,
+    TST = 0x8,
+    TEQ = 0x9,
+    CMP = 0xA,
+    CMN = 0xB,
+    ORR = 0xC,
+    MOV = 0xD,
+    BIC = 0xE,
+    MVN = 0xF,
+}
+
+impl From<u8> for ArmModeAluInstruction {
+    fn from(alu_opcode: u8) -> Self {
+        use ArmModeAluInstruction::*;
+        match alu_opcode {
+            0x0 => AND,
+            0x1 => EOR,
+            0x2 => SUB,
+            0x3 => RSB,
+            0x4 => ADD,
+            0x5 => ADC,
+            0x6 => SBC,
+            0x7 => RSC,
+            0x8 => TST,
+            0x9 => TEQ,
+            0xA => CMP,
+            0xB => CMN,
+            0xC => ORR,
+            0xD => MOV,
+            0xE => BIC,
+            0xF => MVN,
+            _ => unreachable!(),
```

Comment:
Unfortunately, u8 goes up to `0xFF`, so it is not possible for me to cover all cases. 

Btw I wanted to discuss that from my point of view it is better to implement `unreachable!()` than a `try_into().expect()` where possible

Reply:
For now we can keep as is, there are 2 different kind of thoughts, handling errors or fail!
I'm for handling errors in nice way but in this case we are in such situation that will never can fail, I hope, because if something panic probably we are missing some implementation...
If some instruction or conversion can fail we should handle in clean way, but now I don't have a strong answer